core-service/src/test/java/com/crmfoodestablishment/coreservice/service/OrderServiceImplTest.java

```diff
@@ -0,0 +1,52 @@
+package com.crmfoodestablishment.coreservice.service;
+
+import com.crmfoodestablishment.coreservice.dto.order.NewOrderDto;
+import com.crmfoodestablishment.coreservice.entity.Order;
+import com.crmfoodestablishment.coreservice.repository.DishRepository;
+import com.crmfoodestablishment.coreservice.repository.OrderRepository;
+import com.crmfoodestablishment.coreservice.service.exception.NotFoundException;
+import com.crmfoodestablishment.coreservice.service.impl.OrderServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class OrderServiceImplTest {
+
+    private OrderServiceImpl orderService;
+
+    @Mock
+    private OrderRepository orderRepository;
+
+    @Mock
+    private DishRepository dishRepository;
+
+    @Mock
+    private OrderMapper orderMapper;
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.openMocks(this);
+        orderService = new OrderServiceImpl(orderRepository, dishRepository, orderMapper);
+    }
+
+    @Test
+    public void testOrderCreateMethod() {
+        NewOrderDto newOrderDto = new NewOrderDto();
+
+        when(dishRepository.findByUuid(any(UUID.class)).orElseThrow(() -> new NotFoundException("Dish is not found")));
+        when(orderRepository.save(any(Order.class))).thenReturn(new Order());
+
+        UUID result = orderService.createOrder(newOrderDto);
```

Comment:
i don't think it is a correct behavior that u can send empty dto and save it

i think it is a nice place to check if order sum (total price) was calculated correctly
for this purpose u can get trying to use ArgumentCaptor

Reply:
used ArgumentCaptor to check if order sum calculated correctly.